tensorflow/python/eager/backprop.py

```diff
@@ -812,6 +812,9 @@ def watch(self, tensor):
       tensor: a Tensor or list of Tensors.
     """
     for t in nest.flatten(tensor):
+      if not t.dtype.is_floating:
+        logging.vlog(logging.WARN, "The expected dtype of the watched tensor must be "
+        "floating (e.g. tf.float32), got {!r}".format(t.dtype))
```

Comment:
4 spaces for continuation indent, please, e.g.

```python
logging.vlog("..."
    "...")
```

Also (minor nit): I think it's fine to drop "expected" from the error messages.

Reply:
Done, Thanks!